batch-job.go

```diff
@@ -349,6 +423,38 @@ func (adm *AdminClient) ListBatchJobs(ctx context.Context, fl *ListBatchJobsFilt
 	return result, nil
 }
 
+// SupportedBatchJobsHeader is the header key for supported batch jobs included
+// by the server in the list batch jobs response.
+const SupportedBatchJobsHeader = "X-MinIO-Supported-Batch-Jobs"
+
+// ListBatchJobTypes lists the supported batch job types.
+func (adm *AdminClient) ListBatchJobTypes(ctx context.Context) ([]string, error) {
+	resp, err := adm.executeMethod(ctx, http.MethodHead,
+		requestData{
+			relPath: adminAPIPrefix + "/list-jobs",
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+	defer closeResponse(resp)
+
+	if resp.StatusCode != http.StatusOK {
+		if resp.StatusCode == http.StatusUpgradeRequired {
+			// This is the status code returned by community MinIO server (which
+			// does not support HEAD request for this API endpoint). In this
+			// case, we return an empty list of batch job types.
+			return []string{}, nil
+		}
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	supportedBatchJobs := resp.Header.Get(SupportedBatchJobsHeader)
+	supportedBatchJobTypes := strings.Split(supportedBatchJobs, ",")
```

Comment:
Should we validate them against the ones we know? Or is it fine to just return whatever is supported?

(optionally, you can make the a `func (b BatchJobType) Known() bool` that returns if the sdk knows the type)

Reply:
Yes that is checked against `ClientSupportedJobTypes` by client app code (so they could easily display an error instead of the unsupported ones being filtered out by this library).